src/twisted/test/test_dirdbm.py

```diff
@@ -85,21 +91,22 @@ def testDbm(self):
         assert dbitems == copyitems, ".copyTo().items() didn't match: %s != %s" % (repr(dbkeys), repr(copyitems))
 
         d2.clear()
-        assert len(d2.keys()) == len(d2.values()) == len(d2.items()) == 0, ".clear() failed"
+        assert (len(list(d2.keys())) == len(d2.values()) ==
+                len(d2.items()) == 0, ".clear() failed")
         shutil.rmtree(copyPath)
 
-        # delete items
+        # Delete items
         for k, v in self.items:
             del d[k]
             self.assertNotIn(k, d, "key is still in database, even though we deleted it")
-        assert len(d.keys()) == 0, "database has keys"
+        assert len(list(d.keys())) == 0, "database has keys"
```

Comment:
Docstring says that .keys() returns a list. Why is this converted into list here? This seems like a backwards-incompatible change. Can we just do the list() in the original function?